### PR TITLE
[Feature] second correction for exam file upload exercises

### PR DESCRIPTION
**Pull request by SimonEntholzer**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [x] Server: I added `@PreAuthorize` and check the course groups for all new REST Calls (security)
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Server: I documented the Java code using JavaDoc style.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I added multiple integration tests (Jest) related to the features (with a high test coverage)
- [x] Client: I added `authorities` to all new routes and check the course groups for displaying navigation elements (links, buttons)
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context

In [#2833](https://github.com/ls1intum/Artemis/pull/2833) we added file upload exercises to the exam mode. There we didn't yet add the second correction.

### Description
In this PR we will add the second correction round to file-upload exercises in the exam mode, and also adapt the file-upload-assessment dashboard

assessment-dashboard:
image

exercise-assessment-dashboard:
image

file-upload-assessment-dashboard:
image

### Steps for Testing
1. create a new exam with a file upload exercise
2. participate in the exam with a few students
3. assess some of the submissions in the first correction round with a tutor
4. Now check with an instructor in the file-upload-assessment-dashboard (exam -> exercise-groups -> Submissions) if you can see all of the assessed submissions, and if the correct results/scores are loaded when you use "open assessment"
5. test the "new assessment" function with one of the remaining submissions
6. Now, enable the second correction
7. use a different tutor account to assess in the second correction round. Make sure that the results of the first assessment are loaded.
8. After some assessments again double-check in the instructor's file-upload-assessment-dashboard if everything worked fine 

### Test Coverage

file-upload-assessment-dashboard: > 92.54%

### CODE REVIEW

**Review by airelawaleria — COMMENTED** [review]
tested with @simonzilker @marvinchimeka 
We found some issues: 

1. The feedback from the second correction round is not displayed correctly.
image

2. The complaints are also not displayed correctly to the tutors, instead at the bottom of the page, they are displayed above the assessments

**Review by simonzilker — CHANGES_REQUESTED** [review]
What @airelawaleria said. The view when assessing a complaint is not displayed correctl as you can see in the screenshot.
image

**Review by marvinchimeka — COMMENTED** [review]
Confirming what @airelawaleria mentioned.

**Review by JohannesWeiss — COMMENTED** [review]
Can confirm the issues mentioned by @airelawaleria.

**Comment by SimonEntholzer** [comment]
> What @airelawaleria said. The view when assessing a complaint is not displayed correctly as you can see in the screenshot.
> image

This issue should be fixed now, I also made the feedback consistent to those of the text-assessment

**Review by airelawaleria — APPROVED** [review]
tested with  @simonzilker @JohannesWeiss @marvinchimeka the new changes. works as expected

**Review by marvinchimeka — APPROVED** [review]
Works as intended. No issues found

**Review by simonzilker — APPROVED** [review]
Looks good now :)

**Review by JohannesWeiss — DISMISSED** [review]

File: src/main/webapp/app/exercises/file-upload/assess/file-upload-assessment-dashboard.component.ts [hunk 1]
```diff
@@ -46,18 +48,21 @@ export class FileUploadAssessmentDashboardComponent implements OnInit {
...
-            // At least one submission present. Can get exercise from first submission.
-            .then(() => {
-                const exercise = this.submissions[0].participation!.exercise!;
-                FileUploadAssessmentDashboardComponent.verifyFileUploadExercise(exercise);
-                this.exercise = exercise as FileUploadExercise;
+        this.exerciseService
+            .find(exerciseId)
+            .map((exerciseResponse) => {
+                if (exerciseResponse.body!.type !== ExerciseType.FILE_UPLOAD) {
+                    throw new Error('Cannot use Text Assessment Dashboard with non-text Exercise type.');
```
Comment (nicolasruscher) [inline src/main/webapp/app/exercises/file-upload/assess/file-upload-assessment-dashboard.component.ts:55]: I think this should be file upload dashboard, right?

File: src/main/webapp/app/exercises/file-upload/assess/file-upload-assessment.component.html [hunk 2]
```diff
@@ -29,39 +33,29 @@
...
+            <div *ngIf="invalidError" class="col-8 alert alert-danger" role="alert">{{ invalidError | translate }}</div>
+            <jhi-unreferenced-feedback [(feedbacks)]="unreferencedFeedback" (feedbacksChange)="validateAssessment()" [readOnly]="readOnly"></jhi-unreferenced-feedback>
+            <jhi-assessment-general-feedback [(feedback)]="generalFeedback" (feedbackChange)="validateAssessment()" [readOnly]="readOnly"></jhi-assessment-general-feedback>
+        </div>
+    </div>
+</ng-template>
+
+<ng-template #noSubmission>
+    <div class="alert alert-warning text-center mt-4" role="alert">
+        <p jhiTranslate="artemisApp.textAssessment.notFound">We haven't found any new text submission without an assessment, please go back.</p>
```
Comment (nicolasruscher) [inline src/main/webapp/app/exercises/file-upload/assess/file-upload-assessment.component.html:56]: I think this should also be file upload

**Review by nicolasruscher — CHANGES_REQUESTED** [review]
Just some small change requests, other than that it looks good, cool feature!

File: src/main/webapp/app/exercises/file-upload/assess/file-upload-assessment-dashboard.component.ts (hunk 1 above)
Comment (SimonEntholzer) [reply src/main/webapp/app/exercises/file-upload/assess/file-upload-assessment-dashboard.component.ts:55]: uh good catch, I overlooked that.
```suggestion
                    throw new Error('Cannot use File-Upload Assessment Dashboard with non-file-upload Exercise type.');
```

File: src/main/webapp/app/exercises/file-upload/assess/file-upload-assessment.component.html (hunk 2 above)
Comment (SimonEntholzer) [reply src/main/webapp/app/exercises/file-upload/assess/file-upload-assessment.component.html:56]: you are right, I also missed this one
```suggestion
        <p jhiTranslate="artemisApp.fileUploadAssessment.notFound">We haven't found any new text submission without an assessment, please go back.</p>
```

**Review by airelawaleria — APPROVED** [review]
code changes look good:)

**Review by JohannesWeiss — APPROVED** [review]
Re-approve since Code changes lgtm

**Review by LudwigWeide — APPROVED** [review]

**Review by simonzilker — APPROVED** [review]
Re-approve

**Review by marvinchimeka — APPROVED** [review]
Re-approve

**Review by FrankeLukas — APPROVED** [review]
Lgtm. One thing I noticed which is probably not related to this specific PR is that you get an empty page (without any error messages) when you want to assess the next submission if there are no submissions left to assess.

**Review by nicolasruscher — APPROVED** [review]
Re-approve!

**Review by krusche — APPROVED** [review]

**Pull request merged**